src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,811 @@ const ActorInit En_Famos_InitVars = {
+    }
+}
+
+/**
+ * Summary: Regular Famos follows a path until seeing the player.
+ */
+void EnFamos_SetupPathingIdle(EnFamos* this) {
+    if (this->isCalm) {
+        this->currentPathNode++;
+        if (this->currentPathNode == this->pathNodeCount) {
+            this->currentPathNode = 0;
+        }
+    } else {
+        this->isCalm = true;
+    }
+
+    Math_Vec3s_ToVec3f(&this->targetDest, &this->pathPoints[this->currentPathNode]);
+    this->targetYaw = Actor_YawToPoint(&this->actor, &this->targetDest);
+    this->actionFunc = EnFamos_PathingIdle;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFamos_PathingIdle(EnFamos* this, GlobalContext* globalCtx) {
+    EnFamos_UpdateBobbingHeight(this);
+    if (this->isCalm) {
+        Math_Vec3f_Copy(&this->calmPos, &this->actor.world.pos);
+    }
+
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (Math_ScaledStepToS(&this->actor.shape.rot.y, this->targetYaw, 0x200)) {
+        EnFamos_SetupReturnHome(this);
+    }
+}
+
+/**
+ * Summary: Famos lost player; Turning to face back toward home.
+ */
+void EnFamos_SetupTurnHome(EnFamos* this) {
+    this->targetYaw = Actor_YawToPoint(&this->actor, &this->calmPos);
+    Math_Vec3f_Copy(&this->targetDest, &this->calmPos);
+    this->actionFunc = EnFamos_TurnHome;
+    this->actor.speedXZ = 0.0f;
+}
+
+void EnFamos_TurnHome(EnFamos* this, GlobalContext* globalCtx) {
+    EnFamos_UpdateBobbingHeight(this);
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (Math_ScaledStepToS(&this->actor.shape.rot.y, this->targetYaw, 0x200)) {
+        EnFamos_SetupReturnHome(this);
+    }
+}
+
+/**
+ * Summary: Famos has finished rotating toward home; Take off flying in straight line.
+ */
+void EnFamos_SetupReturnHome(EnFamos* this) {
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->actor.world.rot.x = -Actor_PitchToPoint(&this->actor, &this->targetDest);
+    this->actionFunc = EnFamos_ReturnHome;
+}
+
+void EnFamos_ReturnHome(EnFamos* this, GlobalContext* globalCtx) {
+    f32 distanceToHome = Actor_XZDistanceToPoint(&this->actor, &this->targetDest);
+
+    this->actor.shape.rot.y = Actor_YawToPoint(&this->actor, &this->targetDest);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnFamos_UpdateBobbingHeight(this);
+    if (this->isCalm) {
+        Math_Vec3f_Copy(&this->calmPos, &this->actor.world.pos);
+    }
+
+    if (EnFamos_IsPlayerSeen(this, globalCtx)) {
+        EnFamos_SetupAlert(this);
+    } else if (distanceToHome < 20.0f) {
+        if (this->pathPoints != NULL) {
+            EnFamos_SetupPathingIdle(this);
+        } else {
+            EnFamos_SetupStillIdle(this);
+        }
+    } else if (distanceToHome < 40.0f) {
+        Math_StepToF(&this->actor.speedXZ, 0.5f, 0.3f);
+    } else {
+        Math_StepToF(&this->actor.speedXZ, 3.0f, 0.3f);
+    }
+}
+
+/**
+ * Summary: Famos has spotted the player; Delayed by surprise.
```

Comment:
Either this is two sentences with a `.` (although the second one is a fragment), or it's one with a `;` and no capital D.